lib/cronofy/auth.rb

```diff
@@ -22,8 +23,17 @@ def get_token_from_code(code, redirect_uri)
       @client.auth_code.get_token(code, :redirect_uri => redirect_uri).token
     end
 
-    def request
-      @request ||= OAuth2::AccessToken.new(@client, @token)
+    # Public: Refreshes the access token
+    # Returns Hash of token elements to allow client to update in local store for user
+    def refresh!
+      @access_token = access_token.refresh!
```

Comment:
I see your thinking. At a more abstract level the API call is returning an access token so the method returning an `AccessToken` isn't unexpected.

It is possible to [make the OAuth library return a custom type](https://github.com/intridea/oauth2/blob/0ab3c213f7baa43f232eb18fa46301e1a9460f7d/lib/oauth2/client.rb#L128) if we were to extend any of the endpoints.

I think it should be wrapped, at least a little so you don't have to do `token[:access_token]` as the caller, though you've had more experience of calling this code than I have.